### ISSUE TITLE
running in container: failed to detect docker API version

### ISSUE THREAD

**Issue by levlas** [wagoodman/dive#71]
Sorry, if I have misinterpreted the docs (also I skimmed through the issues), but I think the bottom `docker run` command should work
 

$ docker run --rm -it -v /var/run/docker.sock:/var/run/docker.sock -v /usr/bin/docker:/bin/docker wagoodman/dive:latest --version
dive 0.2.0
------------------------------------------------------------------------------------------------------------------------
$ lsb_release -a
\No LSB modules are available.
Distributor ID: Debian
Description:    Debian GNU/Linux 9.5 (stretch)
Release:        9.5
Codename:       stretch
------------------------------------------------------------------------------------------------------------------------
$ /usr/bin/docker version
Client:
 Version:           18.09.0
 API version:       1.39
 Go version:        go1.10.4
 Git commit:        4d60db4
 Built:             Wed Nov  7 00:48:46 2018
 OS/Arch:           linux/amd64
 Experimental:      false

Server: Docker Engine - Community
 Engine:
  Version:          18.09.0
  API version:      1.39 (minimum version 1.12)
  Go version:       go1.10.4
  Git commit:       4d60db4
  Built:            Wed Nov  7 00:16:44 2018
  OS/Arch:          linux/amd64
  Experimental:     false
------------------------------------------------------------------------------------------------------------------------
$ docker run --rm -it -v /var/run/docker.sock:/var/run/docker.sock -v /usr/bin/docker:/bin/docker wagoodman/dive:latest alpine:3.8
Analyzing Image
panic: exit status 127

goroutine 1 [running]:
github.com/wagoodman/dive/utils.DiscoverDockerVersion(0x94e220, 0x0)
        /home/wagoodman/go/src/github.com/wagoodman/dive/utils/docker.go:42 +0x18d
github.com/wagoodman/dive/image.InitializeData(0x7fff6e84df79, 0xa, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, ...)
        /home/wagoodman/go/src/github.com/wagoodman/dive/image/image.go:202 +0xf2
github.com/wagoodman/dive/cmd.analyze(0xc651a0, 0xc4202c7be0, 0x1, 0x1)
        /home/wagoodman/go/src/github.com/wagoodman/dive/cmd/analyze.go:36 +0x119
github.com/spf13/cobra.(*Command).execute(0xc651a0, 0xc4200aa010, 0x1, 0x1, 0xc651a0, 0xc4200aa010)
        /home/wagoodman/go/src/github.com/spf13/cobra/command.go:766 +0x2c1
github.com/spf13/cobra.(*Command).ExecuteC(0xc651a0, 0x2, 0x7f2356cdf6c8, 0x0)
        /home/wagoodman/go/src/github.com/spf13/cobra/command.go:852 +0x30a
github.com/spf13/cobra.(*Command).Execute(0xc651a0, 0xc4201e3f78, 0xc4201e3f58)
        /home/wagoodman/go/src/github.com/spf13/cobra/command.go:800 +0x2b
github.com/wagoodman/dive/cmd.Execute()
        /home/wagoodman/go/src/github.com/wagoodman/dive/cmd/root.go:30 +0x2d
main.main()
        /home/wagoodman/go/src/github.com/wagoodman/dive/main.go:38 +0xad

------------------------------------------------------------------------------------------------------------------------
I have checked the output of this command on my machine, seems good for me:
$ /usr/bin/docker version --format {{.Server.APIVersion}}
1.39

**Comment by sbressey** [comment]
Exactly same error here.
Docker 17.09.1-ce, Fedora 25.

**Comment by wagoodman** [comment]
You are indeed correct, when I fixed #47  I neglected to think of the downstream side effects when running in a container. The problem was assuming that the user's docker client binary would run in the docker image, which is not always the case:
```bash
$ docker run --rm -it -v /var/run/docker.sock:/var/run/docker.sock -v /usr/bin/docker:/bin/docker --entrypoint ''  wagoodman/dive:latest  bash
root@3357f5d08285:/# docker version
docker: error while loading shared libraries: libltdl.so.7: cannot open shared object file: No such file or directory
```

In hindsight it's best to take the API hint from the user via an environment variable instead of trying to detect the API version from the client binary:
```bash
   DOCKER_API_VERSION=1.37 dive ...
``` 
or if you are running with a docker image:
```bash
docker run --rm -it \
    -v /var/run/docker.sock:/var/run/docker.sock \
    -e DOCKER_API_VERSION=1.37
    wagoodman/dive:latest <dive arguments...>
```

This fix should be in the v0.3.0 release
